app/controllers/api/v1/admin/answers_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Admin
+      class AnswersController < Api::V1::Admin::AdminApiController
+        before_action :set_answer, only: %i[show update destroy]
+
+        def index
+          retrieve_answers
+          @pagy, @answers = pagy(@answers, items: params[:per_page])
+        end
+
+        def create
+          question = Question.find(params[:answer][:question_id])
+          @answer = Answer.create!(value: answer_params[:value], question_id: question.id, user_id: current_user.id)
+        end
+
+        def show; end
+
+        def update
+          ActiveRecord::Base.transaction do
+            @answer.update!(answer_params)
+          end
+        end
+
+        def destroy
+          @answer.destroy!
+        end
+
+        private
+
+        def retrieve_answers
+          @answers = Answer.all
+          if params[:question_id].present?
+            question = Question.find(params[:question_id])
+            @answers = question.answers
+          end
+          return unless params[:user_id].present?
+
+          user = User.find(params[:user_id])
+          @answers = user.answers
+        end
+
+        def set_answer
+          @answer = Answer.find(params[:id])
+        end
```

Comment:
no es una buena practica usar metodos con prefijos `get_` o `set_`. lo que se hace en este caso es definir un metodo que se llame `answer` de la siguiente forma:

```ruby
def answer
    @answer ||= Answer.find(params[:id])
end
```

https://jtway.co/how-to-name-variables-and-methods-in-ruby-b4a21fb66c4d en este post habla un poco de eso, por si te interesa.

el operador ||= es un operador que asigna un valor solo si la parte izquierda esta en nil, de esa forma se chachea el valor y solo se hace la query a la base de datos una vez.

los otros cambios que hay que hacer en el controller son sacar los before_action y llamar al metodo `answer` en vez de la variable `@answer`